program/tests/test_create_mint.rs

```diff
@@ -0,0 +1,450 @@
+                ..Default::default()
+            },
+        ),
+        (
+            wrapped_backpointer_address,
+            Account {
+                lamports: 100_000_000,
+                ..Default::default()
+            },
+        ),
+        (unwrapped_mint_address, unwrapped_mint_account.clone()),
+        (
+            system_program::id(),
+            Account {
+                executable: true,
+                ..Default::default()
+            },
+        ),
+        mollusk_svm_programs_token::token2022::keyed_account(),
+    ];
+    let result = mollusk.process_and_validate_instruction(
+        &instruction,
+        accounts,
+        &[
+            Check::success(),
+            // Ensure unwrapped_mint_account remains unchanged
+            Check::account(&unwrapped_mint_address)
+                .data(&unwrapped_mint_account.data)
+                .build(),
+        ],
+    );
+
+    // Assert state of resulting wrapped mint account
+
+    let resulting_wrapped_mint_account = &result.resulting_accounts[0].1;
+    assert_eq!(resulting_wrapped_mint_account.owner, spl_token_2022::id());
+
+    let wrapped_mint_data = Mint::unpack(&resulting_wrapped_mint_account.data).unwrap();
+    assert_eq!(wrapped_mint_data.decimals, MINT_DECIMALS);
+    let expected_mint_authority = get_wrapped_mint_authority(&wrapped_mint_address);
+    assert_eq!(
+        wrapped_mint_data.mint_authority.unwrap(),
+        expected_mint_authority,
+    );
+    assert_eq!(wrapped_mint_data.supply, 0);
+    assert!(wrapped_mint_data.is_initialized);
+    assert_eq!(
+        wrapped_mint_data.freeze_authority.unwrap(),
+        Pubkey::from_str_const(FREEZE_AUTHORITY)
+    );
+
+    // Assert state of resulting backpointer account
+
+    let resulting_backpointer_account = &result.resulting_accounts[1].1;
+    assert_eq!(resulting_backpointer_account.owner, program_id);
+
+    let backpointer = bytemuck::from_bytes::<Backpointer>(&resulting_backpointer_account.data[..]);
+    assert_eq!(backpointer.unwrapped_mint, unwrapped_mint_address);
+}
+
+#[test]
+fn test_successful_token_2022_to_spl_token() {
+    let program_id = spl_token_wrap::id();
+    let mut mollusk = Mollusk::new(&program_id, "spl_token_wrap");
+    mollusk_svm_programs_token::token::add_program(&mut mollusk);
+
+    let unwrapped_mint_address = Pubkey::new_unique();
+    let wrapped_token_program_id = spl_token::id();
+    let wrapped_mint_address =
+        get_wrapped_mint_address(&unwrapped_mint_address, &wrapped_token_program_id);
+    let wrapped_backpointer_address = get_wrapped_mint_backpointer_address(&wrapped_mint_address);
+
+    let instruction = create_mint(
+        &program_id,
+        &wrapped_mint_address,
+        &wrapped_backpointer_address,
+        &unwrapped_mint_address,
+        &wrapped_token_program_id,
+        false,
+    );
+
+    let rent = &mollusk.sysvars.rent;
+    let unwrapped_mint_account = setup_token_2022_mint(rent);
```

Comment:
Can you add at least one extension to the token-2022 mint? The token-2022 tests should show a way to do that: https://github.com/solana-program/token-2022/blob/f69cc989c2a3f2657a6ad6041018468807223f08/program/src/extension/mod.rs#L1890